src/main.rs

```diff
@@ -501,6 +533,34 @@ mod tests {
             assert_eq!(encoded, reencoded);
         }
 
+        #[test]
+        fn reencode_single() {
+            let encoded = cr(SECP_256_R_1, vec![], false);
+            let decoded = CertReq::from_der(&encoded).unwrap();
+            let reencoded = decoded.to_vec().unwrap();
+            assert_eq!(encoded, reencoded);
+        }
+
+        #[tokio::test]
+        async fn kvm_certs_single() {
+            let ext = Extension {
+                extn_id: Kvm::OID,
+                critical: false,
+                extn_value: &[],
+            };
+
+            let request = Request::builder()
+                .method("POST")
+                .uri("/")
+                .header(CONTENT_TYPE, PKCS10)
+                .body(Body::from(cr(SECP_256_R_1, vec![ext], false)))
+                .unwrap();
+
+            let response = app(certificates_state()).oneshot(request).await.unwrap();
+            assert_eq!(response.status(), StatusCode::OK);
+            attest_response(certificates_state(), response, false).await;
+        }
+
```

Comment:
Rather than copy/pasting the tests, it would be better to parameterize them with `rstest`. See: https://docs.rs/rstest/latest/rstest/#creating-parametrized-tests

Reply:
Done